src/tests/system/tests/test_access_control_simple.py

```diff
@@ -0,0 +1,186 @@
+    :customerscenario: False
+    """
+    user1 = provider.user("user1").add(password="Secret123")
+    user2 = provider.user("user2").add(password="Secret123")
+    user3 = provider.user("user3").add(password="Secret123")
+
+    group1 = provider.group("group1").add()
+    group2 = provider.group("group2").add()
+
+    group1.add_member(user1)
+    group1.add_member(user3)
+    group2.add_member(user2)
+    group2.add_member(user3)
+
+    client.sssd.domain["access_provider"] = "simple"
+    client.sssd.domain["simple_allow_groups"] = "group1"
+    client.sssd.domain["simple_deny_groups"] = "group2"
+
+    client.sssd.start()
+
+    assert client.auth.ssh.password("user1", "Secret123"), "User can login!"
+    assert not client.auth.ssh.password("user2", "Secret123"), "User cannot login!"
+    assert not client.auth.ssh.password("user3", "Secret123"), "User cannot login!"
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.importance("critical")
+def test_sssd_simple_allow_and_deny_users_and_groups(client: Client, provider: GenericProvider):
+    """
+    :title: Validate `simple_allow_users`, `simple_deny_users`, `simple_allow_groups`, and `simple_deny_groups`
+    :description: This test checks whether SSSD correctly applies both allow and deny lists
+                for individual users and groups.
+    :setup:
+        1. Create users:
+           - `user1` (explicitly allowed)
+           - `user2` (explicitly allowed)
+           - `user3` (explicitly denied)
+           - `user4` (explicitly denied)
+           - `user5` (not in any list but part of an allowed group)
+           - `user6` (not in any list but part of a denied group)
+        2. Create groups:
+           - `allowed_group` (includes `user5`)
+           - `denied_group` (includes `user6`)
+        3. Configure SSSD with:
+           - `access_provider = simple`
+           - `simple_allow_users = user1, user2`
+           - `simple_deny_users = user3, user4`
+           - `simple_allow_groups = allowed_group`
+           - `simple_deny_groups = denied_group`
+        4. Start SSSD.
+    :steps:
+        1. Attempt login with `user1` (explicitly allowed)
+        2. Attempt login with `user2` (explicitly allowed)
+        3. Attempt login with `user3` (explicitly denied)
+        4. Attempt login with `user4` (explicitly denied)
+        5. Attempt login with `user5` (allowed via group membership)
+        6. Attempt login with `user6` (denied via group membership)
+    :expectedresults:
+        1. `user1` should be able to log in.
+        2. `user2` should be able to log in.
+        3. `user3` should NOT be able to log in.
+        4. `user4` should NOT be able to log in.
+        5. `user5` should be able to log in (due to allowed group membership).
+        6. `user6` should NOT be able to log in (due to denied group membership).
+    :customerscenario: False
+    """
+
+    provider.user("user1").add(password="Secret123")
```

Comment:
```
for i in ["user1", "user2", "user3", "user4", user5", "user6"]:
    provider.user(i).add()
```

Reply:
For me it is hard to decide whether to include logic or not. Should I do the same for test cases I use 2 or 3 users as well?

Reply:
I do whatever has less lines generally and still easy to read. For test case 1-2, it's the same amount of lines so it doesn't matter to me. For here though, I do prefer the logic.